STEERING-COMMITTEE.md

```diff
@@ -1,27 +1,108 @@
-5. Manage the Istio brand to decide which things can be called “Istio” and how that mark can be used in relation to other efforts or vendors.
-6. Resolve any dispute from the Technical Oversight Committee.
+The Steering Committee’s responsibilities include:
+
+1. Establishing rules of governance for the Istio project, including creation
+and ratification of bylaws for the Steering Committee and Technical Oversight
+Committee.
+1. Establishing and enforcing principles to guide the Istio project.
+1. Fostering an environment for a healthy and happy community of developers,
+contributors, vendors, and users.
+1. Defining, evolving, and defending a
+[Code of Conduct](CONTRIBUTING.md#code-of-conduct).
+1. Advising the trademark owner on issues relating to the Istio trademark and
+logo.
+1. Setting marketing and advocacy direction for the project; establishing a
+publishing schedule and vetting content, encouraging and assisting in project
+community members, fostering an ecosystem of vendors, creating content for
+conferences, etc.
+1. Controlling and delegating access to, and establishing processes regarding,
+project resources/assets, including but not limited to artifact repositories,
+build and test infrastructure, web sites and their domains, blogs, and social
+media accounts.
+1. Providing neutral mediation as appropriate to try to resolve non-technical
+disputes that arise as part of the project.
+
+## Membership and voting
+1.  The Steering Committee is structured to allow companies who are most
+invested in the success of the Istio project to participate in business and
+non-technical decision-making.
+1.  All members should abide by the project Code of Conduct.
+1.  There are two types of seats on the Steering Committee: Contributor Seats
+and Community Seats.
+1.  Both Contributor and Community Seats will be appointed beginning in
+July 2020. The appointments for the Contribution and Community seat types will
+expire on staggered dates. After the initial term for both seats, all seats
+will have an annual term.
+1.  Contributor Seat terms expire on January 1 and Community Seat terms expire
+on July 1. If necessary, a company holding a Contributor Seat may change the
+appointed individual at any time during the term.
+1.  No Contributing Company can have more than 5 seats in total on the Steering
+Committee.
+    1.  Subsidiaries, an entity in which another company owns more than 50%
+    of the voting or membership interests; and related companies, companies
+    which are controlled by a common third party entity; and affiliate
+    companies, will be treated together as a single entity, referred to as a
+    Contributing Company.
+1.  There shall be nine **Contributor Seats**. Their allocation is determined
+by the approximate effort and expenditure on the Istio project, as approximated
+by the number of merged pull requests on GitHub over the one year period prior
+to the Contributor Seat assignments:
```

Comment:
We can abstract this language a bit to allow for future enhancements/changes.

I would propose changing "expenditure" to "contributions" unless "expenditure" is added on purpose for direct cost attribution. Here's my suggestion:

The measurements for effort and contributions can change over time via the approval process of the Steering committee members. The current measurement is based on by the number of merged pull requests on GitHub over the one year period prior to the Contributor Seat assignments:

The reason I would suggest we include this is any metrics we come up with can be gamed easily and can have interesting model for incentives. We need to start somewhere and evolve as we learn more.

Reply:
I do like this -- I like the phrasing change (expenditure->contributions) and I also like the new verbiage. Anyone opposed?

Reply:
Could this be "aggregate contributions, including non-code or procedural such as extensive issue triage" - you want to make sure contributor diversity is encouraged

Reply:
The draft said:

> (note: while the project fully embraces that there are ways beyond pull requests to measure input, we anticipate that merged pull requests are a decent proxy for total contribution, which includes reviews, comments, etc)

For the purposes of calculating _company_ contribution, we felt it unlikely that it would be imbalanced such that other contributions were out of alignment with the ratios of participation by PRs.  (I say this as someone whose contributions are not primarily code.) I believe Sean looked at the numbers for other potential metrics, and found them to be about the same.

For an _individual_, we put in the idea of an exclusion for "standing" like Kubernetes has, for voting. For standing for election, the bar to membership seems low enough that an individual (for example, an end user) is likely to meet it through things such as docs, blog or governance contributions.

Reply:
I like that point @nrjpoddar. This was a one of my concerns with proportional seat distribution due to the arguments around calculating contributions.

Reply:
+1 I'm concerned with merged pull requests as well.   I believe it is not an issue today as other contributions would reflect at similar percentage of merged PR, but it could shift in the future.  Have some flexibility like @nrjpoddar suggested would help.  Remember, it is almost impossible to reach consensus to change anything in the charter if we need 80% passing.

Reply:
The point being it should be difficult to change the charter and having a large percentage agreement to change the charter is reasonable/expected.

Reply:
PTAL at the updated language, which addresses this.